app/src/main/java/cz/martykan/forecastie/fragments/AmbiguousLocationDialogFragment.java

```diff
@@ -100,32 +99,27 @@ public void onClick(View view) {
                 final JSONObject sysObject = cityObject.getJSONObject("sys");
 
                 final Calendar calendar = Calendar.getInstance();
-                final String dateMsString = cityObject.getString("dt") + "000";
+                final int dateMsString = cityObject.getInt("dt") * 1000;
```

Comment:
This might need to be a long

Reply:
hmm, that feels clunky. surely there's a date method to do the same?

Reply:
Here is more about the following: We receive a unix time as an int representing seconds, but the method `Calendar.setTimeInMillis()` bellow expects milliseconds. That's why we must multiply by 1000. The problem is, the resulting number is larger than what int allows; therefore a long is needed.

Hmm, now that I see it, the calendar instance was used for setting the weather icon, but now there's now a separate method for that. So we don't need the date here after all :sweat_smile:

Reply:
Fixed in 3629f50fbee9acd5ff180c741355bb80471d1f6b

Reply:
How does ntp, etc. handle it? Given it is precise to the nanoseconds I understand.

Reply:
@robinpaulson I don't think I understand what you mean. Do you mean NTP = Network Time Protocol? In the second paragraph of [the Wikipedia article](https://en.wikipedia.org/wiki/Network_Time_Protocol) it says it is precise to the millisecond.

Reply:
Ah, my bad. I guess I left some information out, I was wondering which variable type NTP uses for storing UNIX time, whether they also use a Long, or some other, time-specific variable.

Reply:
Java itself doesn't seem to have NTP out-of-the-box. The [Apache Commons Net library](https://commons.apache.org/proper/commons-net/) has a [TimeStamp class](https://commons.apache.org/proper/commons-net/apidocs/org/apache/commons/net/ntp/TimeStamp.html) with a higher precision than Java.

Reply:
Further down [the Wikipedia article](https://en.wikipedia.org/wiki/Network_Time_Protocol#Timestamps) it states the NTP timestamp format is more precise than milliseconds, up to a few picoseconds, actually.

But I don't think this is of much relevance for weather forecasts :sweat_smile: